lib/io.js

```diff
@@ -650,6 +650,9 @@ DatumWriter.prototype = {
                     ) {
                 schemaIndex = i;
                 break;
+            } else if (writersSchema.schemas[i].type === 'map' && writersSchema.schemas[i].validate(writersSchema.schemas[i].type, datum)) {
+                schemaIndex = i;
+                break;
```

Comment:
How is this related to the default option?  This seems like it should not be part of this PR.